arviz/tests/test_data.py

```diff
@@ -548,6 +548,24 @@ def test_bad_inference_data():
         InferenceData(posterior=[1, 2, 3])
 
 
+def test_inference_concat_keeps_all_fields():
+    """From failures observed in issue #907"""
+    idata1 = from_dict(posterior={"A": [1, 2, 3, 4]}, sample_stats={"B": [2, 3, 4, 5]})
+    idata2 = from_dict(prior={"C": [1, 2, 3, 4]}, observed_data={"D": [2, 3, 4, 5]})
+
+    idata_c1 = concat(idata1, idata2)
+    idata_c2 = concat(idata2, idata1)
+
+    assert hasattr(idata_c1, "prior")
+    assert hasattr(idata_c2, "prior")
+    assert hasattr(idata_c1, "observed_data")
+    assert hasattr(idata_c2, "observed_data")
+    assert hasattr(idata_c1, "posterior")
+    assert hasattr(idata_c2, "posterior")
+    assert hasattr(idata_c1, "sample_stats")
+    assert hasattr(idata_c2, "sample_stats")

```

Comment:
Also very small comment, the helper function `check_multiple_attrs` can be used, it tries to simplify multiple checks on groups (keys of test_dict) and their variables (values of test_dict). I think the test dict would be something like:

```
test_dict = {
    "posterior": ["A"], 
    "sample_stats": ["B"],
    "prior": ["C"], 
    "observed_data": ["D"],
}
fails_c1 = check_multiple_attrs(test_dict, idata_c1)
assert not fails_c1
fails_c2 = check_multiple_attrs(test_dict, idata_c2)
assert not fails_c2
```

Reply:
Looks like a good improvement, will implement.